packages/plugin-essentials/sources/commands/workspaces/list.ts

```diff
@@ -28,12 +34,24 @@ export default class WorkspacesListCommand extends BaseCommand {
     const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
     const {project} = await Project.find(configuration, this.context.cwd);
 
+    if (configuration.projectCwd === null)
+      throw new UsageError(`This command can only be run from within a Yarn project`);
+
+    const root = await gitUtils.fetchRoot(configuration.projectCwd);
+    const base = root !== null
+      ? await gitUtils.fetchBase(root, {baseRefs: [typeof this.since === `string` ? this.since : await gitUtils.fetchDefaultBranch(root)]})
+      : null;
```

Comment:
I'd prefer this to be consistent with the logic in `yarn version check` (so to use `changesetBaseRefs`). A followup can be to replace all usages of `changesetBaseRefs` (including this one) by `fetchDefaultBranch`, but that would be a major change.

```suggestion
    const root = await gitUtils.fetchRoot(configuration.projectCwd);
    const base = root !== null
      ? await gitUtils.fetchBase(root, {baseRefs: [typeof this.since === `string` ? this.since : configuration.get(`changesetBaseRefs`)]})
      : null;
```

Reply:
Also, would be nice if this wasn't called unless `--since` is set (to avoid crashing if git isn't installed, even if `--since` isn't actually used).

Reply:
Oh yup sure. I wasn't too sure about using `changesetBaseRefs` since the "changeset" part seems specific to the version plugin functionality, but the values seem to be what we're looking for at least. Will update 👍 

Will update that check to conditionally execute too. I was trying to avoid having too many checks for this specific flag, but makes sense to avoid problems if git isn't being used.

Reply:
Fixed in 0fcfcbea37bbd13b1d08ffd1359ce8fc12d96692.